classes/class-wc-connect-nux.php

```diff
@@ -122,12 +122,24 @@ public function is_new_labels_user() {
 		}
 
 		public function register_order_page_labels_pointer( $pointers ) {
-			$dismissed_pointers = explode( ',', (string) get_user_meta( get_current_user_id(), 'dismissed_wp_pointers', true ) );
+			$user_id = get_current_user_id();
+			$dismissed_data = (string) get_user_meta( $user_id, 'dismissed_wp_pointers', true );
+			$dismissed_pointers = explode( ',', $dismissed_data );
 			if ( $dismissed_pointers && in_array( 'wc_services_labels_metabox', $dismissed_pointers ) ) {
 				return $pointers;
 			}
 
-			if ( $this->is_new_labels_user() && $this->shipping_label->should_show_meta_box() ) {
+			// If the user is not new to labels, we should just dismiss this pointer
+			if ( ! $this->is_new_labels_user() ) {
+				if ( strlen( $dismissed_data ) ) {
+					$dismissed_data .= ',';
+				}
+				$dismissed_data .= 'wc_services_labels_metabox';
```

Comment:
> when $dismissed_data is empty

Was thinking we could use the `$dismissed_pointers` array without string manipulation on `$dismissed_data`. Something like:

```
if ( ! $this->is_new_labels_user() ) {
  $dismissed_pointers[] = 'wc_services_labels_metabox';
  update_user_meta( $user_id, 'dismissed_wp_pointers', implode( $dismissed_pointers ) );
}
```

> it's hard coded to read from $_POST

Was thinking of refreshing the page with a post request, but can see how that can get even more complicated since we'll need to catch that post data somewhere.